app/controllers/category_exercises_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CategoryExercisesController < ApplicationController
+  before_action :set_category
+
+  def new
+    @timed_exercises = current_user.timed_exercises.all_except(@category.timed_exercises)
+    @rep_exercises = current_user.repetition_exercises.all_except(@category.repetition_exercises)
+  end
+
+  def create
+    @category.add_exercise_in_category(exercise_params)
+    redirect_to categories_url, notice: "Successfully added exercise in #{@category.name}"
+  end
+
+  private
+
+  def exercise_params
+    params.require(:category_exercise).permit(repetition: [], timed: []).tap do |param|
```

Comment:
what is #tap?

Reply:
umm ano siya, like diba yang line 17, imagine it in this way:
`params.require(:category_exercise).permit(repetition: [], timed: [])`

The above line of code returns a hash.

If we want to modify a hash, we need to do this:
![image](https://user-images.githubusercontent.com/94411241/183314661-f3d5b4c9-073b-4a4e-8385-046ac8989dec.png)

Instead of the above, nag .tap method lang ako para mas malinis.
Basically, pinapasa mo yung params hash mismo don sa `|param|`. 
So in the code sa .tap, `|param|` is actually yung mismong params. It's just for better readability since you don't need to assign variable to return the modified hash. You just need to pass a block on that hash, and then you know na yung ginawa mong operations sa loob ng block, ayun dapat yung irereturn na hash where there are operations applied on it.

It doesn't matter naman that much since may debate din about the readability sa tap method, others like it, others doesn't hehehehe.

I think mas better tong explanation hahahaha

![image](https://user-images.githubusercontent.com/94411241/183314743-af835b34-ea4e-458a-86ca-b3ad89bdb41f.png)